### PR TITLE
Report Known Xapi OOVPA issue list

### PR DESCRIPTION
**Pull request by jarupxx**
Plus reregisterd obsolete OOVPA

### CODE REVIEW

**Comment by PatrickvL** [comment]
Good work, as always, thanks!
However, I'm a bit reluctant to merge this in - could @RadWolfie take a look perhaps?

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl [hunk 1]
```diff
@@ -61,36 +101,31 @@
...
-	// REGISTER_OOVPAS(XCalculateSignatureBegin, PATCH, 3911),
 	// REGISTER_OOVPAS(XCalculateSignatureBegin, PATCH, 4627),
 	// REGISTER_OOVPAS(XCalculateSignatureBeginEx, PATCH, 4627), // +s, not necessary? (from 4627, 5028's comment)
 	// REGISTER_OOVPAS(XCalculateSignatureEnd, PATCH, 4627), // s+ (from 4627, 5028's comment)
 	// REGISTER_OOVPAS(XCalculateSignatureUpdate, PATCH, 4627),
-	// REGISTER_OOVPAS(XInputGetDeviceDescription, PATCH, 4831), // NOT XInputGetDeviceDescription (from 4627, 5028's comment)
-	// REGISTER_OOVPAS(XapiBootDash, PATCH, 3911), // obsolete (from 4721's comment)
+	// REGISTER_OOVPAS(ReadFileEx, DISABLED, 3911),// Has exact asm codes _WriteFileEx@20
+	// REGISTER_OOVPAS(WriteFileEx, DISABLED, 3911), // Has exact asm codes _ReadFileEx@20
+	// REGISTER_OOVPAS(lstrcmpiW, DISABLED, 3911), // TODO: necessary XREF _xCompareStringW@20
```
[the remark this answers is not in the document]
Comment (RadWolfie) [reply src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: `xCompareStringW` and `xCompareStringA` has many difference asm lines. We can make XREF OOVPA for both in order to get accurate reading of `lstrcmpiW` and `lstrcmpiA` detection.

@jarupxx, you don't have to add both XREF OOVPAs right away. We can still merge it, except need to update the comment to knowledge exact same asm codes in comment above.

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl [hunk 2]
```diff
@@ -61,36 +101,31 @@
...
+	REGISTER_OOVPAS(CloseHandle, DISABLED, 3911),
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: "Has exact same asm codes as _ResetEvent@4"

File: src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl [hunk 3]
```diff
@@ -1232,6 +1262,97 @@ OOVPA_NO_XREF(RaiseException, 3911, 7)
 OOVPA_END;
 
 // ******************************************************************
+// * SwitchToThread
+// ******************************************************************
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl]: Please add "// Generic OOVPA as of 3911 and newer." without the quote here.

File: src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl [hunk 4]
```diff
@@ -1232,6 +1262,97 @@ OOVPA_NO_XREF(RaiseException, 3911, 7)
...
+        { 0x16, 0x89 },
+        { 0x17, 0x45 },
+        // { 0x18, 0xE4 }, 3911 0xE4 vs 5558 0xE0
+        
+        { 0x1F, 0x89 },
+OOVPA_END;
+
+// ******************************************************************
+// * MoveFileA
+// ******************************************************************
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl]: Please add "// Generic OOVPA as of 3911 and newer." without the quote here.

File: src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl [hunk 5]
```diff
@@ -1232,6 +1262,97 @@ OOVPA_NO_XREF(RaiseException, 3911, 7)
...
+        { 0x0E, 0x95 },
+        { 0x0F, 0xC1 },
+        { 0x10, 0x8B },
+        { 0x11, 0xC1 },
+        { 0x12, 0xC3 },
+OOVPA_END;
+
+// ******************************************************************
+// * XapiThreadStartup
+// ******************************************************************
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl]: Please add "// Generic OOVPA as of 3911 and newer." without the quote here.

File: src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl [hunk 6]
```diff
@@ -1232,6 +1262,97 @@ OOVPA_NO_XREF(RaiseException, 3911, 7)
...
+        { 0x08, 0x35 },
+
+        { 0x1F, 0x8D },
+        { 0x93, 0xC2 },
+        { 0x94, 0x08 },
+OOVPA_END;
+
+// ******************************************************************
+// * XapiFiberStartup
+// ******************************************************************
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.1.0.3911.inl]: Please add "// Generic OOVPA as of 3911 and newer." without the quote here.

File: src/CxbxKrnl/HLEDataBase/Xapi.1.0.4039.inl [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+// *  (c) 2017 jarupxx
+// *
+// *  All rights reserved
+// *
+// ******************************************************************
+
+
+// ******************************************************************
+// * XCalculateSignatureBegin
+// ******************************************************************
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.1.0.4039.inl]: Please add "// Generic OOVPA as of 4039 and newer." without the quote here.

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl [hunk 8]
```diff
@@ -61,36 +101,31 @@
...
+	REGISTER_OOVPAS(CreateMutex, DISABLED, 3911), // Too High Level (from 3911's comment)
+	REGISTER_OOVPAS(CreateThread, DISABLED, 3911), // Too High Level (from 3911's comment)
+	REGISTER_OOVPAS(ExitThread, DISABLED, 3911),//
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: ExitThread OOVPA has a weak OV total. Might be best to include Offset 0x00, 0x00B, and 0x00C, to strengthen a bit better.
@jarupxx, can you do this task before we merge it?

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl (hunk 2 above)
Comment (RadWolfie) [reply src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: Also we can't register this OOPA in v2 database for duplicate functions serve different purpose.

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl (hunk 8 above)
Comment (jarupxx) [reply src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: I can work on this task after 8 hours, but I would appreciate it if you do work instead.

Comment (jarupxx) [reply src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: All right, ExitThread OOVPA include all offsets.

**Comment by RadWolfie** [comment]
Um, I cannot squash this pull request. Nor even can do a rebase.

@PatrickvL & @LukeUsher, can't you guys leave the options for pull request alone? As in leave them all enable. I notice some "toggling" merge options in the past week has occurred.

**Comment by PatrickvL** [comment]
I see no restrictions in this github project settings. Luke has probably enabled everything again. Are your still having difficulty merging?

**Comment by RadWolfie** [comment]
Thanks

**Pull request merged**